### DISCUSSION TITLE
vim.surround repeat with . operator does not select whole word

### DISCUSSION THREAD

**Discussion by sourcevault** [NeoVintageous/NeoVintageous#937] [General]
The repeat operation on vim.surround does not do what I was expecting it to do.

https://github.com/NeoVintageous/NeoVintageous/assets/20347647/8017c6d2-0c52-4697-bce8-29ce12891798

I am not sure if that is a bug or if this plugin needs to be supported:

https://github.com/tpope/vim-repeat

**clearing vim register**

There is a command in vim (`q` + `<register name>` + `q`) to clear a named registry, it basically just uses the marco function to assign a registry a blank value.

**Comment by gerardroche** [comment]
The reason is the `w` word motion doesn't include special characters like `*`. The WORD motion `W` does. So `ysiW.` will work because `W` includes special characters. 

The other option is moving one character to the right so `w` doesn't have to match the `*`: `ysiwl.`.

The vim-repeat doesn't help with this. Surround and Repeat have the same quirk.

**Reply by sourcevault** [reply]
Hi @gerardroche, thank you for clarifying my mistake 🙏.

**Reply by sourcevault** [reply]
![vim_lines](https://github.com/NeoVintageous/NeoVintageous/assets/20347647/69c5ca84-caed-4853-9d98-656f5b6ac426)

I created a helpful diagram for anybody new to neovint and vim.

**Reply by gerardroche** [reply]
Very good.
